corehq/motech/dhis2/repeaters.py

```diff
@@ -416,10 +419,12 @@ def send_request(self, repeat_record, payload):
             )
             raise
 
+    def _wrap_schema_attrs(self, couch_object):
+        setattr(couch_object, 'dhis2_entity_config', Dhis2EntityConfig.wrap(getattr(self, 'dhis2_entity_config')))
```

Comment:
if `dhis2_entity_config` is just a dict then how does this work:

https://github.com/dimagi/commcare-hq/blob/8f37a226e23c5bc009dd05be2077f7258c541dc0/corehq/motech/dhis2/repeaters.py#L395

The same question applies to the other OptionValue attributes that are `dict` (`SQLDhis2Repeater.dhis2_config`,  `SQLOpenmrsRepeater.openmrs_config`)

Reply:
Also why do you need to use `getattr` here? If the class doesn't have the attribute `getattr` will return None which I think will cause an error with `.wrap`

Reply:
> if dhis2_entity_config is just a dict then how does this work:


This won't work. This should be self.dhis2_entity_config['case_configs'].
Currently, the work to update SQLRepeater functions and make them compatible is [work-in-progress](https://github.com/dimagi/commcare-hq/compare/ap/repeater-migration/move-reads-1) 
It would be fixed when we would move the reads.

> Also why do you need to use getattr here? If the class doesn't have the attribute getattr will return None which I think will cause an error with .wrap


`_wrap_schema_attrs` is overridden only defined for those classes which have respective attributes. So it would not be called for any class which do not have the respective attributes.

Reply:
> `_wrap_schema_attrs` is overridden only defined for those classes which have respective attributes. So it would not be called for any class which do not have the respective attributes.

So then are you saying that `getattr` isn't needed?

Reply:
Yep. You are correct 😄  
Updated https://github.com/dimagi/commcare-hq/pull/31956/commits/7966699878a0a3d8eae5aaad82625c4dec66e79b